Sources/ScrollingTabController.swift

```diff
@@ -202,18 +196,32 @@ open class ScrollingTabController: UIViewController, UIScrollViewDelegate, UICol
     }
 
     func configureViewControllers() {
-        for item in items {
-            let child = item.controller
-            child.willMove(toParentViewController: nil)
-            child.view.removeFromSuperview()
-            child.removeFromParentViewController()
-            item.container.removeFromSuperview()
-        }
-        
         for viewController in viewControllers {
             items.append(TabItem(addTabContainer(), viewController))
         }
     }
+
+    public func injectInitialViewControllers(_ viewControllers: [UIViewController]) {
+        guard tabControllersView != nil else { return }
+        self.viewControllers = viewControllers
+        configureViewControllers()
+    }
+
+    public func appendViewController(_ newViewControllers: [UIViewController]) {
+        var inserts: [IndexPath] = []
+        for index in 0..<newViewControllers.count {
+            inserts.append(IndexPath(item: index + viewControllers.count, section: 0))
+        }
+        for vc in newViewControllers {
+            items.append(TabItem(addTabContainer(), vc))
+        }
+        viewControllers += newViewControllers
+        tabView.collectionView.performBatchUpdates({
+            self.tabView.collectionView.insertItems(at: inserts)
+        }, completion: { _ in
+            self.tabView.panToPercentage(self.scrolledPercentage)
```

Comment:
For `ScrollingTabView.TabSizing = .fitViewFrameWidth` we probably need to recalculate item widths 😄